ITR/data/base_providers.py

```diff
@@ -516,46 +573,53 @@ def _compute_missing_historic_ei(self, companies, historic_data):
                 projections = [IProjection(year=year, value=value) for year, value in results.items()
                                if year in range(self.projection_controls.BASE_YEAR, self.projection_controls.TARGET_YEAR+1)]
-                scope_projections[scope] = ICompanyEIProjections(ei_metric={'units': units}, projections=projections)
+                scope_projections[scope_name] = ICompanyEIProjections(ei_metric=units, projections=projections)
             if scope_projections['S1'] and scope_projections['S2'] and not scope_projections['S1S2']:
                 results = scope_dfs['S1'] + scope_dfs['S2']
                 units = f"{results.values[0].u:~P}"
                 projections = [IProjection(year=year, value=value) for year, value in results.items()
                                if year in range(self.projection_controls.BASE_YEAR, self.projection_controls.TARGET_YEAR+1)]
-                scope_projections['S1S2'] = ICompanyEIProjections(ei_metric={'units': units}, projections=projections)
+                scope_projections['S1S2'] = ICompanyEIProjections(ei_metric=units, projections=projections)
             # FIXME: do we really need to do this?  We're going to migrate S3 to S1S2 and ignore S1S2S3...
             if scope_projections['S1S2'] and scope_projections['S3'] and not scope_projections['S1S2S3']:
                 results = scope_dfs['S1S2'] + scope_dfs['S3']
                 units = f"{results.values[0].u:~P}"
                 projections = [IProjection(year=year, value=value) for year, value in results.items()
                                if year in range(self.projection_controls.BASE_YEAR, self.projection_controls.TARGET_YEAR+1)]
-                scope_projections['S1S2S3'] = ICompanyEIProjections(ei_metric={'units': units}, projections=projections)
+                scope_projections['S1S2S3'] = ICompanyEIProjections(ei_metric=units, projections=projections)
             company.projected_intensities = ICompanyEIProjectionsScopes(**scope_projections)
 
     def _standardize(self, intensities: pd.DataFrame) -> pd.DataFrame:
         # When columns are years and rows are all different intensity types, we cannot winsorize
         # Transpose the dataframe, winsorize the columns (which are all coherent because they belong to a single variable/company), then transpose again
-        intensities = intensities.T
-        for col in intensities.columns:
-            s = intensities[col]
-            ei_units = f"{s.loc[s.first_valid_index()].u:~P}"
-            if s.notnull().any():
-                with warnings.catch_warnings():
-                    warnings.simplefilter("ignore")
-                    intensities[col] = s.map(lambda x: Q_(np.nan, ei_units)
-                    if x.m is np.nan or x.m is pd.NA else x).astype(f"pint[{ei_units}]")
+        na_intensities = intensities.apply(lambda x: x.isna().all(), axis=1)
+        intensities = intensities[~na_intensities].T
+        if ITR.HAS_UNCERTAINTIES:
```

Comment:
seems like we lost an `else` piece.
we remove lines 550-558, but don't replace them with anything

Reply:
Actually line 610 replaces all the exploration of lines 551-558, and line 611 gives us the transposed meaningful values to standardize (line 550).  This is one of the features of the new pint-pandas 0.3 (which has proper pd.na for Quantity as an ExtensionDtype in an ExtensionArray).
What's tricky here is that pd.na only works on PintArray data, so we need to be sure that our incoming row data is properly PintArray'd.  Sadly we then have to dequantify the data for the quantile function, but we reconstruct PintArrays afterward...